hibike/lib/hibike/hibike_message.cpp

```diff
@@ -81,90 +81,69 @@ int read_message(message_t *msg) {
 
 }
 
-int send_subscription_response(hibike_uid_t* uid, uint16_t delay) {
+int send_subscription_response(uint16_t params, uint16_t delay, hibike_uid_t* uid) {
   message_t msg;
   msg.messageID = SUBSCRIPTION_RESPONSE;
   msg.payload_length = 0;
 
-  int status = 0;
+  int status = append_payload(&msg, (uint8_t*) &params, sizeof(params));
+  
+  status += append_payload(&msg, (uint8_t*) &delay, sizeof(delay));
+  
   status += append_payload(&msg, (uint8_t*) &uid->device_type, UID_DEVICE_BYTES);
   status += append_payload(&msg, (uint8_t*) &uid->year, UID_YEAR_BYTES);
   status += append_payload(&msg, (uint8_t*) &uid->id, UID_ID_BYTES);
 
-  status += append_payload(&msg, (uint8_t*) &delay, sizeof(delay));
 
   if (status != 0) {
     return -1;
   }
   return send_message(&msg); 
 }
 
-
-int send_data_update(uint8_t* data, uint8_t payload_length) {
+//deprecated but not really
+int send_data_update(uint16_t params) { 
   message_t msg;
-  msg.messageID = DATA_UPDATE;
-  msg.payload_length = 0;
-  int status = append_payload(&msg, data, payload_length);
-  
-  if (status != 0) {
+  msg.messageID = DEVICE_DATA;
+  msg.payload_length = 2;
+  for (uint16_t count = 0; (params >> count) > 0; count++) {
+      if (params & (1<<count)){
+        int bytes_written = device_data_update((uint8_t) count, &msg.payload[msg.payload_length], (size_t) msg.payload_length);
+        if(bytes_written){
+          msg.payload_length += bytes_written;
+        }
+        else{
+          params &= ~(1<<count);
+        }
+      }
+  }
+
+  *((uint16_t *) &msg.payload[0]) = params;
+
+  if (msg.payload_length > MAX_PAYLOAD_SIZE) {
+    // toggleLED();
     return -1;
+  } else {
+    // send_data_update(params, hibikeBuff.payload,hibikeBuff.payload_length); //what does this do?
+    return send_message(&msg); //what does this do?
```

Comment:
is that comment "what does this do" still relevant? 
If so, i'd like it answered.